### ISSUE TITLE
'di' moves cursor during op-pending mode if text is concealed and `d<tab>` is mapped in normal mode

### ISSUE THREAD

**Issue by justinmk** [neovim/neovim#1356]
Noticed in c28adf15e6c2079c732bb77fb99c50b80a4d7fe2. Didn't bisect yet.

Steps to reproduce:

```
nvim -u NONE

:filetype on
:syntax on
:onoremap ia :echo 'foo'<cr>
:nnoremap d<tab> <c-w>c
" This defines `|` as a concealed char
:set ft=help
:set conceallevel=2
:set concealcursor=nc
```
1. In the buffer, enter this text: `|foo| barbaz zub`
2. On the character "a", invoke `di`. 
   - **Bug:** During this operator-pending mode, cursor jumps forward 2 characters. (Note that there are 2 concealed `|` chars to the left.)

**Comment by rev112** [comment]
Is the issue still present? I tried but couldn't reproduce it.

**Comment by justinmk** [comment]
Just reproduced it on 6270d431aaeed71e7a8782411f36409ab8e0ee35. Did you follow all the steps?

**Comment by rev112** [comment]
I'm doing the following (please correct me):

1. Launch `nvim`
2. `:set concealcursor=nc`
3. `:set ft=help`
4. Go to INSERT mode, enter `|foo| barbaz zub`
5. Return to NORMAL, go to the first "a" character (or the second, makes no difference)
6. Invoke `di`

Nothing happens.

**Comment by justinmk** [comment]
Sorry, there were (many) missing steps since I didn't reproduce with `nvim -u NONE`. Updated the steps.

The important part is really `nnoremap d<tab>`.

Note that Vim doesn't have this issue, so it's probably a Neovim regression.

**Comment by rev112** [comment]
Still no luck in reproducing :(
Maybe I just don't get how `di` is related to `d<tab>`...
Is there something besides `di` which cause this behavior?

**Comment by lucc** [comment]
I can only reproduce it if I add `set showcmd`. I use the following test-init.vim and test.in files:
```vim
filetype on
syntax on
onoremap ia :echo 'foo'<cr>
nnoremap d<tab> <c-w>c
" This defines `|` as a concealed char
set ft=help
set conceallevel=2
set concealcursor=nc
set showcmd
```
```
|foo| barbaz zab
```
Then I start nvim from current master (45b378259ebc66530cfa2a3fdb2504f17a185a22) with `VIMRUNTIME=./runtime build/bin/nvim -u test-init.vim test.in` and press `fadi`. Then I see the cursor jump. If I remove the last line from the init file I do not see the cursor jump. *But in both cases the behavior in vanilla vim is the same.*

So is this really a bug?

**Comment by justinmk** [comment]
@lucc Thanks for that! Confirmed on Vim 7.4.728. Still seems like a bug to me...

**Comment by zeertzjq** [comment]
Seems fixed in v0.4.0.
